crates/wasi-http/src/common/table.rs

```diff
@@ -0,0 +1,114 @@
+// Based on:
```

Comment:
Does this really belong in `wasi-http` crate or should it be living alongside the existing https://docs.wasmtime.dev/api/wasi_common/table/struct.Table.html in `wasi_common`? Should there be a `wasi_common::component::table::Table` if this is component-model specific?

Reply:
yes, when wasmtime supports wasi preview2, the wasi common here will most likely contain the table (following the preview2-prototyping approach). until then, i just copied this over and made some small tweaks.

Reply:
we are using the shared table as provided by `wasmtime-wasi`.